lib/init.js

```diff
@@ -2,15 +2,31 @@
 
 module.exports = init
 
+var path = require('path')
 var log = require('npmlog')
 var npm = require('./npm.js')
+var npx = require('libnpx')
 var initJson = require('init-package-json')
 var output = require('./utils/output.js')
 var noProgressTillDone = require('./utils/no-progress-while-running').tillDone
 
 init.usage = 'npm init [--force|-f|--yes|-y]'
 
 function init (args, cb) {
+  if (args.length) {
+    var NPM_PATH = path.resolve(__dirname, '../bin/npm-cli.js')
+    var initerName = args[0]
+    var packageName = /^@[^/]+$/.test(initerName)
+      ? initerName + '/create'
+      : initerName.replace(/^(@[^/]+\/)?/, '$1create-')
+
```

Comment:
👆 In addition to following yarn's rules, name expansion also supports transforming `@scope` into `@scope/create`.